cli/ask/ask.go

```diff
@@ -0,0 +1,147 @@
+package ask
+
+import (
+	"context"
+	"flag"
+	"os"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/cli/arg"
+	"github.com/buildbuddy-io/buildbuddy/cli/log"
+	"github.com/buildbuddy-io/buildbuddy/cli/login"
+	"github.com/buildbuddy-io/buildbuddy/cli/storage"
+	"github.com/buildbuddy-io/buildbuddy/server/util/grpc_client"
+	"github.com/buildbuddy-io/buildbuddy/server/util/uuid"
+	"google.golang.org/grpc/metadata"
+
+	bbspb "github.com/buildbuddy-io/buildbuddy/proto/buildbuddy_service"
+	supb "github.com/buildbuddy-io/buildbuddy/proto/suggestion"
+)
+
+var (
+	flags  = flag.NewFlagSet("ask", flag.ContinueOnError)
+	openai = flags.Bool("openai", false, "If true, use openai endpoint")
+)
+
+const (
+	invocationIDFlagName = "invocation_id"
+	besBackendFlagName   = "bes_backend"
+)
+
+var (
+	usage = `
+usage: bb ` + flags.Name() + ` [--openai]
+
+Asks for suggestions about the previous invocation.
+`
+)
+
+func HandleAsk(args []string) (int, error) {
+	command, idx := arg.GetCommandAndIndex(args)
+	if command != "ask" && command != "wtf" && command != "huh" {
+		return -1, nil
+	}
+
+	if idx != 0 {
+		log.Debugf("Unexpected flag: %s", args[0])
+		return 1, nil
+	}
+
+	if err := arg.ParseFlagSet(flags, args[idx+1:]); err != nil {
+		if err == flag.ErrHelp {
+			log.Print(usage)
+			return 1, nil
+		}
+		return 1, err
+	}
+
+	lastIID, err := getPreviousFlag(invocationIDFlagName)
+	if lastIID == "" || err != nil {
+		log.Printf("Couldn't find the previous invocation.")
+		return 1, err
+	}
+
+	req := &supb.GetSuggestionRequest{
+		InvocationId: string(lastIID),
+	}
+
+	if *openai {
+		req.Service = supb.SuggestionService_OPENAI
+	}
+
+	apiKey, err := storage.ReadRepoConfig("api-key")
+	if err != nil {
+		exitCode, err := login.HandleLogin([]string{"login"})
+		if exitCode > 0 || err != nil {
+			return exitCode, err
+		}
+		apiKey, err = storage.ReadRepoConfig("api-key")
+		if apiKey == "" || err != nil {
+			return 1, err
+		}
+	}
+
+	ctx := metadata.AppendToOutgoingContext(context.Background(), "x-buildbuddy-api-key", apiKey)
+
+	lastBackend, err := getPreviousFlag(besBackendFlagName)
+	if lastBackend == "" || err != nil {
+		log.Printf("The previous invocation didn't have the --bes_backend= set.")
+		return 1, err
+	}
+
+	if !strings.HasPrefix(lastBackend, "grpc://") && !strings.HasPrefix(lastBackend, "grpcs://") {
+		lastBackend = "grpcs://" + lastBackend
+	}
+
+	conn, err := grpc_client.DialTarget(lastBackend)
+	if err != nil {
+		return 1, err
+	}
+	client := bbspb.NewBuildBuddyServiceClient(conn)
+	res, err := client.GetSuggestion(ctx, req)
+	if err != nil {
+		return 1, err
+	}
+
+	for _, s := range res.Suggestion {
+		log.Printf(s)
+	}
+
+	return 0, nil
+}
+
+// TODO(siggisim): Move this out of the ask package if we want to save more flags.
+func SaveFlags(args []string) []string {
```

Comment:
do you want to save these in a workspace specific location or are you OK with them being user-level?

Reply:
Added a hash of the workspace directory to the filename